include/swift/Sema/ConstraintSystem.h

```diff
@@ -4838,19 +4839,38 @@ class ConstraintSystem {
       });
     }
 
-    unsigned getNumDefaultableBindings() const {
-      return isDirectHole()
-                 ? 1
-                 : llvm::count_if(Bindings,
-                                  [](const PotentialBinding &binding) {
-                                    return binding.isDefaultableBinding();
-                                  });
+    unsigned getNumViableDefaultableBindings() const {
+      if (isDirectHole())
+        return 1;
+
+      // We might want to consider adding this as a field to `PotentialBindings`
+      // and collect this information as new bindings become available but,
+      // since we are moving towards incremental model, which implies that
+      // `PotentialBindings` would stay alive for a long time, it's not
+      // immediately clear whether storage overhead of keeping this set just
+      // for ranking is worth it.
+      llvm::SmallPtrSet<CanType, 4> discoveredTypes;
+      for (const auto &binding : Bindings)
```

Comment:
After looking at this again, I think I can make it work without having to use a separate set.

Reply:
Ok, done!